### ISSUE TITLE
Build error on Arch Linux, detecting LibreSSL when it's not installed

### ISSUE THREAD

**Issue by ShapeShifter499** [dogecoin/dogecoin#1370]
This is with dogecoin 1.10

I'm getting a compile error, it's complaining about LibreSSL but I don't have it installed as far as I can tell. The pacman packager doesn't show it installed. 

```
checking for miniupnpc/miniwget.h... yes
checking for main in -lminiupnpc... yes
checking miniupnpc/miniupnpc.h usability... yes
checking miniupnpc/miniupnpc.h presence... yes
checking for miniupnpc/miniupnpc.h... yes
checking for main in -lminiupnpc... (cached) yes
checking miniupnpc/upnpcommands.h usability... yes
checking miniupnpc/upnpcommands.h presence... yes
checking for miniupnpc/upnpcommands.h... yes
checking for main in -lminiupnpc... (cached) yes
checking miniupnpc/upnperrors.h usability... yes
checking miniupnpc/upnperrors.h presence... yes
checking for miniupnpc/upnperrors.h... yes
checking for main in -lminiupnpc... (cached) yes
checking for QtCore QtGui QtNetwork... yes
checking for QtTest... yes
checking for QtDBus... yes
checking for moc-qt4... /usr/bin/moc-qt4
checking for uic-qt4... /usr/bin/uic-qt4
checking for rcc-qt4... /usr/bin/rcc-qt4
checking for lrelease-qt4... /usr/bin/lrelease-qt4
checking for lupdate-qt4... /usr/bin/lupdate-qt4
checking whether to build Dogecoin Core GUI... yes (Qt4)
checking for boostlib >= 1.20.0... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Program_Options library is available... yes
checking for exit in -lboost_program_options... yes
checking whether the Boost::Thread library is available... yes
checking for exit in -lboost_thread... yes
checking whether the Boost::Chrono library is available... yes
checking for exit in -lboost_chrono... yes
checking whether the Boost::Unit_Test_Framework library is available... yes
checking for dynamic linked boost test... yes
checking for libssl... yes
checking for libcrypto... yes
checking for protobuf... yes
checking for libqrencode... yes
checking for RAND_egd in -lcrypto... no
configure: error: Detected LibreSSL: This is NOT supported, and may break consensus compatibility!
```

**Comment by ShapeShifter499** [comment]
The patches from the Debian Dogecoin package fix my issues https://sources.debian.net/src/dogecoin/1.10.0-5/debian/patches

**Comment by ShapeShifter499** [comment]
I get a new build issue after the above patches. 

```
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
Building LevelDB ...
make[3]: Entering directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/leveldb'
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c helpers/memenv/memenv.cc -o helpers/memenv/memenv.o
rm -f libmemenv.a
/usr/bin/ar -rs libmemenv.a helpers/memenv/memenv.o
/usr/bin/ar: creating libmemenv.a
make[3]: Leaving directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/leveldb'
Building LevelDB ...
make[3]: Entering directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/leveldb'
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/builder.cc -o db/builder.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/c.cc -o db/c.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/db_impl.cc -o db/db_impl.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/db_iter.cc -o db/db_iter.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/dbformat.cc -o db/dbformat.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/dumpfile.cc -o db/dumpfile.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/filename.cc -o db/filename.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/log_reader.cc -o db/log_reader.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/log_writer.cc -o db/log_writer.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/memtable.cc -o db/memtable.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/repair.cc -o db/repair.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/table_cache.cc -o db/table_cache.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/version_edit.cc -o db/version_edit.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/version_set.cc -o db/version_set.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c db/write_batch.cc -o db/write_batch.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/block.cc -o table/block.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/block_builder.cc -o table/block_builder.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/filter_block.cc -o table/filter_block.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/format.cc -o table/format.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/iterator.cc -o table/iterator.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/merger.cc -o table/merger.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/table.cc -o table/table.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/table_builder.cc -o table/table_builder.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c table/two_level_iterator.cc -o table/two_level_iterator.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/arena.cc -o util/arena.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/bloom.cc -o util/bloom.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/cache.cc -o util/cache.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/coding.cc -o util/coding.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/comparator.cc -o util/comparator.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/crc32c.cc -o util/crc32c.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/env.cc -o util/env.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/env_posix.cc -o util/env_posix.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/env_win.cc -o util/env_win.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/filter_policy.cc -o util/filter_policy.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/hash.cc -o util/hash.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/histogram.cc -o util/histogram.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/logging.cc -o util/logging.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/options.cc -o util/options.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c util/status.cc -o util/status.o
g++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -I. -I./include -std=c++0x -fno-builtin-memcmp -pthread -DOS_LINUX -DLEVELDB_PLATFORM_POSIX -DLEVELDB_ATOMIC_PRESENT -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong  -Wstack-protector -fstack-protector-all -fPIC -D_FORTIFY_SOURCE=2 -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -D__STDC_LIMIT_MACROS -c port/port_posix.cc -o port/port_posix.o
rm -f libleveldb.a
/usr/bin/ar -rs libleveldb.a db/builder.o db/c.o db/db_impl.o db/db_iter.o db/dbformat.o db/dumpfile.o db/filename.o db/log_reader.o db/log_writer.o db/memtable.o db/repair.o db/table_cache.o db/version_edit.o db/version_set.o db/write_batch.o table/block.o table/block_builder.o table/filter_block.o table/format.o table/iterator.o table/merger.o table/table.o table/table_builder.o table/two_level_iterator.o util/arena.o util/bloom.o util/cache.o util/coding.o util/comparator.o util/crc32c.o util/env.o util/env_posix.o util/env_win.o util/filter_policy.o util/hash.o util/histogram.o util/logging.o util/options.o util/status.o port/port_posix.o
/usr/bin/ar: creating libleveldb.a
make[3]: Leaving directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/leveldb'
make[3]: Entering directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/secp256k1'
  CC       src/libsecp256k1_la-secp256k1.lo
  CCLD     libsecp256k1.la
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: Leaving directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src/secp256k1'
  CXX      wallet/libbitcoin_wallet_a-crypter.o
  CXX      wallet/libbitcoin_wallet_a-db.o
  CXX      wallet/libbitcoin_wallet_a-rpcdump.o
  CXX      wallet/libbitcoin_wallet_a-rpcwallet.o
  CXX      wallet/libbitcoin_wallet_a-wallet.o
  CXX      wallet/libbitcoin_wallet_a-wallet_ismine.o
  CXX      wallet/libbitcoin_wallet_a-walletdb.o
  CXX      libbitcoin_wallet_a-dogecoin.o
  AR       libbitcoin_wallet.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXXLD    dogecoind
  CXX      dogecoin_cli-bitcoin-cli.o
  CXX      libbitcoin_cli_a-rpcclient.o
  CXX      libbitcoin_cli_a-dogecoin.o
  AR       libbitcoin_cli.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXXLD    dogecoin-cli
  CXX      dogecoin_tx-bitcoin-tx.o
  CXXLD    dogecoin-tx
Generated test/data/script_valid.json.h
Generated test/data/base58_keys_valid.json.h
Generated test/data/base58_encode_decode.json.h
Generated test/data/base58_keys_invalid.json.h
Generated test/data/script_invalid.json.h
Generated test/data/tx_invalid.json.h
Generated test/data/tx_valid.json.h
Generated test/data/sighash.json.h
Generated test/data/alertTests.raw.h
  CXX      test/test_test_dogecoin-arith_uint256_tests.o
  CXX      test/test_test_dogecoin-alert_tests.o
  CXX      test/test_test_dogecoin-allocator_tests.o
  CXX      test/test_test_dogecoin-auxpow_tests.o
  CXX      test/test_test_dogecoin-base32_tests.o
  CXX      test/test_test_dogecoin-base58_tests.o
  CXX      test/test_test_dogecoin-base64_tests.o
  CXX      test/test_test_dogecoin-bip32_tests.o
  CXX      test/test_test_dogecoin-bloom_tests.o
  CXX      test/test_test_dogecoin-checkblock_tests.o
  CXX      test/test_test_dogecoin-Checkpoints_tests.o
  CXX      test/test_test_dogecoin-coins_tests.o
  CXX      test/test_test_dogecoin-compress_tests.o
  CXX      test/test_test_dogecoin-crypto_tests.o
  CXX      test/test_test_dogecoin-DoS_tests.o
  CXX      test/test_test_dogecoin-dogecoin_tests.o
  CXX      test/test_test_dogecoin-getarg_tests.o
  CXX      test/test_test_dogecoin-hash_tests.o
  CXX      test/test_test_dogecoin-key_tests.o
  CXX      test/test_test_dogecoin-main_tests.o
  CXX      test/test_test_dogecoin-mempool_tests.o
  CXX      test/test_test_dogecoin-miner_tests.o
  CXX      test/test_test_dogecoin-mruset_tests.o
  CXX      test/test_test_dogecoin-multisig_tests.o
  CXX      test/test_test_dogecoin-netbase_tests.o
  CXX      test/test_test_dogecoin-pmt_tests.o
  CXX      test/test_test_dogecoin-policyestimator_tests.o
  CXX      test/test_test_dogecoin-pow_tests.o
  CXX      test/test_test_dogecoin-rpc_tests.o
  CXX      test/test_test_dogecoin-sanity_tests.o
  CXX      test/test_test_dogecoin-scheduler_tests.o
  CXX      test/test_test_dogecoin-script_P2SH_tests.o
  CXX      test/test_test_dogecoin-script_tests.o
  CXX      test/test_test_dogecoin-scriptnum_tests.o
In file included from test/scriptnum_tests.cpp:5:0:
test/bignum.h:26:24: error: invalid use of incomplete type ‘BIGNUM {aka struct bignum_st}’
 class CBigNum : public BIGNUM
                        ^~~~~~
In file included from /usr/include/openssl/bn.h:32:0,
                 from test/bignum.h:16,
                 from test/scriptnum_tests.cpp:5:
/usr/include/openssl/ossl_typ.h:80:16: note: forward declaration of ‘BIGNUM {aka struct bignum_st}’
 typedef struct bignum_st BIGNUM;
                ^~~~~~~~~
In file included from test/scriptnum_tests.cpp:5:0:
test/bignum.h: In constructor ‘CBigNum::CBigNum()’:
test/bignum.h:31:21: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
                     ^
test/bignum.h: In copy constructor ‘CBigNum::CBigNum(const CBigNum&)’:
test/bignum.h:36:21: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
                     ^
test/bignum.h:37:30: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘BIGNUM* BN_copy(BIGNUM*, const BIGNUM*)’
         if (!BN_copy(this, &b))
                              ^
test/bignum.h:39:31: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘void BN_clear_free(BIGNUM*)’
             BN_clear_free(this);
                               ^
test/bignum.h: In member function ‘CBigNum& CBigNum::operator=(const CBigNum&)’:
test/bignum.h:46:30: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘BIGNUM* BN_copy(BIGNUM*, const BIGNUM*)’
         if (!BN_copy(this, &b))
                              ^
test/bignum.h: In destructor ‘CBigNum::~CBigNum()’:
test/bignum.h:53:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘void BN_clear_free(BIGNUM*)’
         BN_clear_free(this);
                           ^
test/bignum.h: In constructor ‘CBigNum::CBigNum(long long int)’:
test/bignum.h:56:49: error: ‘BN_init’ was not declared in this scope
     CBigNum(long long n)          { BN_init(this); setint64(n); }
                                                 ^
test/bignum.h: In constructor ‘CBigNum::CBigNum(const std::vector<unsigned char>&)’:
test/bignum.h:60:21: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
                     ^
test/bignum.h: In member function ‘int CBigNum::getint() const’:
test/bignum.h:66:38: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘long unsigned int BN_get_word(const BIGNUM*)’
         BN_ULONG n = BN_get_word(this);
                                      ^
test/bignum.h:67:33: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_negative(const BIGNUM*)’
         if (!BN_is_negative(this))
                                 ^
In file included from test/scriptnum_tests.cpp:5:0:
test/bignum.h: In member function ‘void CBigNum::setint64(int64_t)’:
test/bignum.h:115:37: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(pch, p - pch, this);
                                     ^
test/bignum.h: In member function ‘void CBigNum::setvch(const std::vector<unsigned char>&)’:
test/bignum.h:130:46: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(&vch2[0], vch2.size(), this);
                                              ^
test/bignum.h: In member function ‘std::vector<unsigned char> CBigNum::getvch() const’:
test/bignum.h:135:50: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         unsigned int nSize = BN_bn2mpi(this, NULL);
                                                  ^
test/bignum.h:139:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         BN_bn2mpi(this, &vch[0]);
                                ^
test/bignum.h: In function ‘const CBigNum operator+(const CBigNum&, const CBigNum&)’:
test/bignum.h:153:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_add(BIGNUM*, const BIGNUM*, const BIGNUM*)’
     if (!BN_add(&r, &a, &b))
                           ^
test/bignum.h: In function ‘const CBigNum operator-(const CBigNum&, const CBigNum&)’:
test/bignum.h:161:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_sub(BIGNUM*, const BIGNUM*, const BIGNUM*)’
     if (!BN_sub(&r, &a, &b))
                           ^
test/bignum.h: In function ‘const CBigNum operator-(const CBigNum&)’:
test/bignum.h:169:43: error: cannot convert ‘CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_negative(const BIGNUM*)’
     BN_set_negative(&r, !BN_is_negative(&r));
                                           ^
test/bignum.h: In function ‘bool operator==(const CBigNum&, const CBigNum&)’:
test/bignum.h:173:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator==(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) == 0); }
                                                                                   ^
test/bignum.h: In function ‘bool operator!=(const CBigNum&, const CBigNum&)’:
test/bignum.h:174:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator!=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) != 0); }
                                                                                   ^
test/bignum.h: In function ‘bool operator<=(const CBigNum&, const CBigNum&)’:
test/bignum.h:175:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator<=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) <= 0); }
                                                                                   ^
test/bignum.h: In function ‘bool operator>=(const CBigNum&, const CBigNum&)’:
test/bignum.h:176:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator>=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) >= 0); }
                                                                                   ^
test/bignum.h: In function ‘bool operator<(const CBigNum&, const CBigNum&)’:
test/bignum.h:177:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator<(const CBigNum& a, const CBigNum& b)  { return (BN_cmp(&a, &b) < 0); }
                                                                                   ^
test/bignum.h: In function ‘bool operator>(const CBigNum&, const CBigNum&)’:
test/bignum.h:178:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 inline bool operator>(const CBigNum& a, const CBigNum& b)  { return (BN_cmp(&a, &b) > 0); }
                                                                                   ^
make[2]: *** [Makefile:6408: test/test_test_dogecoin-scriptnum_tests.o] Error 1
make[2]: Leaving directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src'
make[1]: *** [Makefile:6726: all-recursive] Error 1
make[1]: Leaving directory '/home/raansu/My-AUR-git/dogecoin-qt/src/dogecoin-1.10.0/src'
make: *** [Makefile:627: all-recursive] Error 1
```

**Comment by Thermi** [comment]
That looks like it's not compiling with openssl 1.1.0.e, which I mentioned on the page of the package on the AUR. Unless that's patched, I can't update it.

**Comment by langerhans** [comment]
@Thermi Current master compiles against 1.0.1l. Also if those warnings are not specific to Dogecoin code, then they would probably be fixed upstream and we're going to pull them in with the work on 1.14.

**Comment by Thermi** [comment]
They are specific to the dogecoin code. I think you can read german (They are printed in german, even if I pass LANG=C to `makepkg`), so I'm attaching the output of `LANG=C makepkg -f 2> errrors` to this comment.
[errors.txt](https://github.com/dogecoin/dogecoin/files/1068362/errors.txt)

Btw, when's the next release going to happen and what commits/patches do I need to apply to fix the errors (and warnings)?

**Comment by langerhans** [comment]
None of the warnings in that file are specific to Dogecoin code. They all reference code from upstream bitcoin. Plan for the next release is roughly this summer. Work has been started on the 1.14-dev branch. Once this release is done I'd also advise to actually let the integrated dependency builder build all the required libs so you always get the right version guaranteed. But that is probably a later issue.

**Comment by Thermi** [comment]
Where are the dependencies going to be installed to? If you're going to ship them with the software (package), you have to install them to the application specific directory.

**Comment by langerhans** [comment]
They are not installed. They are compiled from source to be then linked into the final dogecoin-qt build. They can be removed afterwards if you wanted to. See here: https://github.com/dogecoin/dogecoin/blob/master/depends/README.md
Alas, this does not fully work on current master. We have a working branch but would need to actually merge it... Sorry to be a pain to support :|

**Comment by Thermi** [comment]
Statically linked? TBH, that approach is pretty bad, because it works around the OS mechanism for software updates, so users will end up with software that has publicly known vulns because of old third party libs. I'd rather pin the versions using the dependencies on the package level.

**Comment by langerhans** [comment]
Right, then pinning sounds fair. The version we have used to compile the 1.10 release are available in the depends/packages subdir: https://github.com/dogecoin/dogecoin/tree/master/depends/packages  
Given 1.14 will bring us on par with Bitcoin 0.14 I hoping most of these issues will be resolved then.

**Comment by Thermi** [comment]
I can't force users to downgrade packages, so I'll wait for the next release and then introduce pinning. Do you have a complete list of dependencies with the versions?

**Comment by Thermi** [comment]
Linking against openssl 1.0 does not fix those problems:
Can we have a fix for that or recommendations?
```
  CXX      qt/qt_libbitcoinqt_a-sendcoinsentry.o
qt/paymentrequestplus.cpp: In member function 'bool PaymentRequestPlus::getMerchant(X509_STORE*, QString&) const':
qt/paymentrequestplus.cpp:164:27: error: 'EVP_MD_CTX_new' was not declared in this scope
         EVP_MD_CTX *ctx = EVP_MD_CTX_new();
                           ^~~~~~~~~~~~~~
qt/paymentrequestplus.cpp:164:27: note: suggested alternative: 'EVP_MD_CTX_init'
         EVP_MD_CTX *ctx = EVP_MD_CTX_new();
                           ^~~~~~~~~~~~~~
                           EVP_MD_CTX_init
qt/paymentrequestplus.cpp:170:13: error: 'EVP_MD_CTX_free' was not declared in this scope
             EVP_MD_CTX_free(ctx);
             ^~~~~~~~~~~~~~~
qt/paymentrequestplus.cpp:170:13: note: suggested alternative: 'EVP_MD_CTX_create'
             EVP_MD_CTX_free(ctx);
             ^~~~~~~~~~~~~~~
             EVP_MD_CTX_create
qt/paymentrequestplus.cpp:173:2: error: 'EVP_MD_CTX_free' was not declared in this scope
  EVP_MD_CTX_free(ctx);
  ^~~~~~~~~~~~~~~
qt/paymentrequestplus.cpp:173:2: note: suggested alternative: 'EVP_MD_CTX_create'
  EVP_MD_CTX_free(ctx);
  ^~~~~~~~~~~~~~~
  EVP_MD_CTX_create
make[2]: *** [Makefile:4791: qt/qt_libbitcoinqt_a-paymentrequestplus.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

**Comment by xslugx** [comment]
I am also having this issue on Ubuntu 14.x.x, Is this an issue with the openssl package itself or how dogecoin references the package?

**Comment by ReverseControl** [comment]
The original issue was, in fact, resolved. @xslugx @ShapeShifter499 Feel free to open a new issue if your problems still persist.
